potato/server_utils/schemas/likert.py

```diff
@@ -1,112 +1,140 @@
-    label_requirement = annotation_scheme.get("label_requirement")
-    if label_requirement and label_requirement.get("required"):
+    if annotation_scheme.get("label_requirement", {}).get("required"):
         validation = "required"
+        logger.debug(f"Setting required validation for {annotation_scheme['name']}")
 
+    # Generate scale points
     for i in range(1, annotation_scheme["size"] + 1):
-
-        label = "scale_" + str(i)
-        name = annotation_scheme["name"] + ":::" + label
-        class_name = annotation_scheme["name"]
-
+        label = f"scale_{i}"
+        name = f"{annotation_scheme['name']}:::{label}"
         key_value = str(i % 10)
 
-        # if the user wants us to add in easy key bindings
-        if "sequential_key_binding" in annotation_scheme and annotation_scheme["sequential_key_binding"] == True and annotation_scheme["size"] < 10:
-            key2label[key_value] = label
-            label2key[label] = key_value
-            key_bindings.append((key_value, class_name + ": " + key_value))
+        # Handle key bindings for scales with less than 10 points
+        if (annotation_scheme.get("sequential_key_binding")
+            and annotation_scheme["size"] < 10):
+            key_bindings.append((key_value, f"{annotation_scheme['name']}: {key_value}"))
+            logger.debug(f"Added key binding '{key_value}' for point {i}")
 
-        # In the collapsed version of the likert scale, no label is shown.
+        # Format label content - show numbers if displaying_score is enabled
         label_content = str(i) if annotation_scheme.get("displaying_score") else ""
-        tooltip = ""
-
-        # displaying the label content in a different line if it is not empty
-        if label_content != "":
-            line_break = "<br>"
-        else:
-            line_break = ""
-        # schematic += \
-        #        ((' <li><input class="%s" type="radio" id="%s" name="%s" value="%s" onclick="onlyOne(this)">' +
-        #         '  <label for="%s" %s>%s</label></li>')
-        #         % (class_name, label, name, key_value, name, tooltip, label_content))
-
-        schematic += (
-            (
-                ' <li><input class="{class_name}" type="radio" id="{id}" name="{name}" value="{value}" onclick="onlyOne(this)" validation="{validation}">'
-                + ' {line_break} <label for="{label_for}" {label_args}>{label_text}</label></li>'
-            )
-        ).format(
-            class_name=class_name,
-            id=name,
-            name=name,
-            value=key_value,
-            validation=validation,
-            line_break=line_break,
-            label_for=name,
-            label_args=tooltip,
-            label_text=" " + label_content,
-        )
-
-    # allow annotators to choose bad_text label
-    bad_text_schematic = ""
+        line_break = "<br>" if label_content else ""
+
+        # Generate radio input for each scale point
+        schematic += f"""
```

Comment:
I didn't know you could do multi-line template strings like this. That is very cool!